services/workflow/workflow-service/src/main/java/com/milesight/beaveriot/rule/components/serviceinvocation/ServiceInvocationComponent.java

```diff
@@ -0,0 +1,53 @@
+package com.milesight.beaveriot.rule.components.serviceinvocation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.milesight.beaveriot.context.api.EntityValueServiceProvider;
+import com.milesight.beaveriot.context.api.ExchangeFlowExecutor;
+import com.milesight.beaveriot.context.integration.model.ExchangePayload;
+import com.milesight.beaveriot.rule.annotations.RuleNode;
+import com.milesight.beaveriot.rule.api.ProcessorNode;
+import com.milesight.beaveriot.rule.constants.RuleNodeType;
+import com.milesight.beaveriot.rule.support.SpELExpressionHelper;
+import lombok.Data;
+import org.apache.camel.Exchange;
+import org.apache.camel.spi.UriParam;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author loong
+ * @date 2024/12/23 11:34
+ */
+@RuleNode(value = "serviceInvocation", type = RuleNodeType.ACTION, description = "Service Invocation")
+@Data
+public class ServiceInvocationComponent implements ProcessorNode<Exchange> {
+
+    @UriParam(prefix = "bean")
+    private String serviceEntity;
+    @UriParam(javaType = "java.util.Map", prefix = "bean")
+    private Map<String, Object> serviceParams;
+
+    @Autowired
+    EntityValueServiceProvider entityValueServiceProvider;
+    @Autowired
+    private ExchangeFlowExecutor exchangeFlowExecutor;
+
+    @Override
+    public void processor(Exchange exchange) {
+        List<String> entitiesVariables = (List<String>) SpELExpressionHelper.resolveExpression(exchange, Collections.singletonList(this.serviceEntity));
+        Map<String, JsonNode> entityValues = entityValueServiceProvider.findValuesByKeys(entitiesVariables);
```

Comment:
The service call should be to build the ExchangePayload directly with serviceParams

Reply:
fixed